src/identifiers.md

```diff
@@ -60,4 +63,37 @@ See the [HIR chapter][hir-map] for more detailed information.
 
 ## In the MIR
 
-**TODO**
+- [`BasicBlock`] identifies a *basic block*. It points to an instance of
+  [`BasicBlockData`].
+
+- [`Local`] identifies a local variable in a function. Its associated data is in
+  [`LocalDecl`].
+
+- [`Field`] identifies a struct's, union's, or enum variant's field. It is used
+  as a "projection" in [`Place`].
+
+- [`SourceScope`] identifies a name scope in the original source code. Used for
+  diagnostics and for debuginfo in debuggers.
+
+- [`Promoted`] identifies a promoted constant within another item (related to
+  const evaluation). Note: it is unique only locally within the item;
```

Comment:
If I understand correctly, the basic unit in MIR is `Body`?
```suggestion
  const evaluation). Note: it is unique only locally within the body;
```

Reply:
Hmm, well the promoted constant *is* a `Body`. I think all items are `Body`s, but not all `Body`s are items. Basically, a `Body` represents a chunk of code that's executed together, which may be extracted from a function's `Body` as a promoted constant. So in this case, I think item is correct. Though I would like to hear from @oli-obk if I'm right here :)

Reply:
I immediately thought this is correct, but writing up my response I realized that inline-constants (so the new `const { 42 }` expressions), are not items, but is an `AnonConst`, which has a `DefId`. Though we can just treat these as "anonymous const item" and then everything fits again.

So... the general problem is that you need a `DefId` + `Promoted` pair to identify a specific promoted. So if you have something which does not have `DefId` but has a `Body`, then you cannot have `Promoted`s in it. Thus, you cannot have promoteds within other promoteds, even if a promoted could refer to another promoted from another item or even the same item.

Reply:
@oli-obk So should I change the documentation somehow? If so, I'm not fully sure what you are requesting I add.

Reply:
Not sure. @LeSeulArtichaut does my comment above make it clearer? If so, do you have any suggestion on how to best put this into the text?

Reply:
I think we could either have a footnote with your comment, just leave it as is, or add something like
```suggestion
  const evaluation). Note: it is unique only locally within the item, so it should always be associated with a `DefId`;
```

Reply:
That snippet sounds like a good middle ground between explaining it in full and just leaving it as it is.